MekHQ/src/mekhq/gui/StratconTab.java

```diff
@@ -160,30 +189,173 @@ private void updateCampaignState() {
             .append("<br/>")
             .append(campaignState.getBriefingText());
         
-        // avoid confusing users by showing strategic objectives when there are none to show
-        if (!campaignState.strategicObjectivesBehaveAsVPs()) {
-            sb.append("<br/>Strategic Objectives: ").append(campaignState.getStrategicObjectiveCompletedCount())
-                .append("/").append(campaignState.getPendingStrategicObjectiveCount());
-        }
-        
         sb.append("<br/>Victory Points: ").append(campaignState.getVictoryPoints())
             .append("<br/>Support Points: ").append(campaignState.getSupportPoints())
             .append("<br/>Deployment Period: ").append(currentTDI.track.getDeploymentTime())
             .append(" days")
             .append("</html>");
         
         campaignStatusText.setText(sb.toString());
+        
+        objectiveStatusText.setText(getStrategicObjectiveText(campaignState));
+    }
+    
+    /**
+     * Builds strategic objective text, appropriately appending details
+     * if the objectives are not "collapsed".
+     */
+    private String getStrategicObjectiveText(StratconCampaignState campaignState) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("<html>")
+            .append(buildShortStrategicObjectiveText(campaignState));
+        
+        if (objectivesCollapsed) {
+            sb.append(" [+] ");
+        } else {
+            sb.append(" [-]<br/>")
+                .append(buildStrategicObjectiveText(campaignState));
+        }
+        
+        sb.append("</html>");
+        
+        return sb.toString();
+    }
+    
+    /**
+     * Builds strategic objective one-liner summary
+     */
+    private String buildShortStrategicObjectiveText(StratconCampaignState campaignState) {
+        int completedObjectives = 0, desiredObjectives = 0;
+        
+        for (StratconTrackState track : campaignState.getTracks()) {
+            for (StratconStrategicObjective objective : track.getStrategicObjectives()) {
+                desiredObjectives++;
+                
+                if (objective.isObjectiveCompleted(track)) {
+                    completedObjectives++;
+                }
+            }
+        }
+        
+        StringBuilder sb = new StringBuilder();
+        if (completedObjectives >= desiredObjectives) {
+            sb.append("<span color='green'>");
+        } else {
+            sb.append("<span color='red'>");
+        }
+        
+        // special logic for non-independent command clauses
+        if (campaignState.getContract().getCommandRights() <= Contract.COM_LIAISON) {
+            desiredObjectives++;
+            
+            if (campaignState.getVictoryPoints() > 0) {
+                completedObjectives++;
+            }
+        }
+
+        sb.append("Strategic objectives: " + completedObjectives + "/" + desiredObjectives + " completed</span>");
+        return sb.toString();
```

Comment:
Can probably merge

Reply:
meaning?

Reply:
sb.append.toString

Reply:
No need to make that line longer, I think.